### DISCUSSION TITLE
Extensions for the ModelBuilder fluent interface

### DISCUSSION THREAD

**Discussion by mchlrch** [eclipse-rdf4j/rdf4j#3467] [Ideas]
I'm frequently using the `ModelBuilder` fluent interface. To streamline my code, I wrote a few additional convenience methods for adding collections and for skipping over nullable values.

```
addEach(Map<IRI, Object> propertyValues)
addEach(String predicate, Collection<?> objectValues)
addNullable(String predicate, Object object)
```

ATM these methods are in a [utils library](https://github.com/mchlrch/rdf4j-utils2). What do you think about adding them to the `ModelBuilder` in the rdf4j codebase?

I would create a PR, if there is a good chance to get it merged.

**Comment by abrokenjester** [comment]
That definitely looks useful. One thing I'm curious about: does `addEach` add a single property with an RDF collection as its value, or does it add loads of properties, each with one value? 

Another thought: instead of having `addNullable` should we consider making the standard add method accept null?

In any case: a contribution would be welcome! Before you get started, please have a look at the [contributor guidelines](https://github.com/eclipse/rdf4j/blob/main/.github/CONTRIBUTING.md), in particular the points on how to sign the Eclipse ECA. Let me know if you need any help with any of it.

**Reply by fkleedorfer** [reply]
Concerning the addNullable: +1, but I'd prefer the normal method to reject null, and call the method that accepts null (but then does nothing), to be called addIfPresent.
